### PR TITLE
[ISSUE #5845] Improve the code quality.

### LINKED ISSUE #5845: InitUtils.java The method of initNamespaceForNaming has redundant codes
微信截图_20210525174855
微信截图_20210525174925

### PR DESCRIPTION
**Pull request by mask616**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

FIX #5845

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by brotherlu-xcq — APPROVED** [review]

**Review by KomachiSion — CHANGES_REQUESTED** [review]
Why remove this logic？

**Comment by mask616** [comment]
> Why remove this logic？

这里是一段冗余代码， 在TenantUtil.getUserTenantForAns()里边，已经调用了`System.getProperty("ans.namespace")`.
image

image

**Comment by KomachiSion** [comment]
这两个请求不一样 `System.getProperty`是从JVM参数中取
`System.getenv`是从环境变量中取

File: client/src/main/java/com/alibaba/nacos/client/naming/utils/InitUtils.java [hunk 1]
```diff
@@ -58,15 +58,6 @@ public static String initNamespaceForNaming(Properties properties) {
         if (Boolean.parseBoolean(isUseCloudNamespaceParsing)) {
             
             tmpNamespace = TenantUtil.getUserTenantForAns();
-            tmpNamespace = TemplateUtils.stringEmptyAndThenExecute(tmpNamespace, new Callable<String>() {
-                @Override
-                public String call() {
-                    String namespace = System.getProperty(SystemPropertyKeyConst.ANS_NAMESPACE);
```
Comment (mask616) [inline client/src/main/java/com/alibaba/nacos/client/naming/utils/InitUtils.java:64]: @KomachiSion 在我删除的这段代码中， 也是从System.getProperty获取的。在TenantUtil.getUserTenantForAns()中也是通过System.getProperty获取的。所以， 我删除的应该是一段冗余代码。

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/naming/utils/InitUtils.java:64]: 删除`TenantUtil.getUserTenantForAns()`可能不合适，因为还可能取USER_TENANT。
缺少日志可以把`stringEmptyAndThenExecute`中的日志移动到`TenantUtil.getUserTenantForAns()`调用之后打印。

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: client/src/main/java/com/alibaba/nacos/client/naming/utils/InitUtils.java (hunk 1 above)
Comment (mask616) [reply client/src/main/java/com/alibaba/nacos/client/naming/utils/InitUtils.java:64]: > 删除`TenantUtil.getUserTenantForAns()`可能不合适，因为还可能取USER_TENANT。
> 缺少日志可以把`stringEmptyAndThenExecute`中的日志移动到`TenantUtil.getUserTenantForAns()`调用之后打印。

@KomachiSion 这个PR可以通过吗？如果可以的话， 那我在`TenantUtil.getUserTenantForAns()`调用之后打印一下日志。

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/naming/utils/InitUtils.java:64]: 你先修改。改完再看看。

**Comment by KomachiSion** [comment]
This PR author has no more response and code changes. We think close it first. If some one want to do again. Please submit a new PR.

**Pull request closed without merging**